src/main/groovy/pl/allegro/tech/build/axion/release/domain/NextVersionMarker.groovy

```diff
@@ -9,22 +10,34 @@ class NextVersionMarker {
 
     private static final ReleaseLogger logger = ReleaseLogger.Factory.logger(NextVersionMarker)
 
-    private final ScmService repository
+    private final ScmService repositoryService
+    private final VersionConfig versionConfig
 
-    NextVersionMarker(ScmService repository) {
-        this.repository = repository
+    NextVersionMarker(ScmService repositoryService, VersionConfig config) {
+        this.repositoryService = repositoryService
+        this.versionConfig = config
     }
 
     void markNextVersion(NextVersionProperties nextVersionRules, TagProperties tagRules) {
-        if (nextVersionRules.nextVersion == null) {
-            throw new IllegalArgumentException("No next version specified! Use -Prelease.version to set next version.")
+
+        def version = nextVersionRules.nextVersion
```

Comment:
the logic here is a bit hidden - it would be great to have explicit branches for *version* vs *incrementer*, so that the precedence is explicit:

```
String nextVersion = null
if (nextVersionRules.nextVersion) {
    nextVersion = ....
} else {
    String currentVersion = ...
    nextVersion = ....
}

```

Also all of the code uses explicit types, i would like to continue this way instead of introducing `def`.

Reply:
I would also change the name of local variable from `version` to `nextVersion`.